### DISCUSSION TITLE
get length of stringt with color syntax

### DISCUSSION THREAD

**Discussion by mathisloevenich** [Textualize/rich#1559] [Q&A] [answered]
Strings like:

`
color_str = "[red]alert[/red]"
`

do obviously have a length of more than the actual printed length.
It should be possible to get the raw length `= 5`

I have looked into the docs and feel like rich.text should be able to make the job.

However I did not find explicit documentation for this use case.

**Comment by willmcgugan** [comment, marked as answer]
That's correct, you can use the Text class.

```python
>>> from rich.text import Text
>>> text = Text.from_markup("[red]alert[/red]")
>>> len(text)
5
```

**Reply by mathisloevenich** [reply]
Nice thanks 👍
